knowledge-base/adr/0031-resilience-options.md

```diff
@@ -170,17 +169,15 @@ This is done the follwing way:
 ```ts
 const [timeout, circuitBreaker] = resilience();
 function myTimeout(inOut: MiddlewareInOut) {
-  if (inOut.context.category === 'user-defined') {
-    const wrapped = (args: any[]) =>
-      exitChain
-        ? fn(args)
-        : Promise.race([fn(args), timeoutPromise<T>(longerTimeout)]); //custom super long timeout
-    return {
-      fn: wrapped,
-      context,
-      exitChain
-    };
-  }
+  const wrapped = (args: any[]) =>
+    exitChain
```

Comment:
[q] Where is the `exitChain` variable coming from? 

[pp] Assuming this is code that the users write, I would not like them to handle `exitChain` themselves but rather give them a function `exitMiddlewareChain()` that will interrupt the chain of middlewares (so it just does not execute the following methods rather than having to implement the ternary yourself).

Reply:
I mentioned this on the other PR as well. I am not a fan of the exitChain either. I would rather reuse the express middleware concept with `next()`.